[pt-br] Add /docs/tasks/administer-cluster/coredns.md (#47656)
kubernetes/website: content/pt-br/docs/tasks/administer-cluster/coredns.md

---
reviewers:
title: Usando o CoreDNS para Descoberta de Serviços
min-kubernetes-server-version: v1.9
content_type: task
weight: 380
---

<!-- overview -->
Essa página descreve o processo de atualização do CoreDNS e como instalar o CoreDNS ao invés de kube-dns.

## {{% heading "prerequisites" %}}

{{< include "task-tutorial-prereqs.md" >}} {{< version-check >}}

<!-- steps -->

## Sobre o CoreDNS

[CoreDNS](https://coredns.io) é um servidor DNS flexível e extensível
que pode servir como Kubernetes cluster DNS.
Como o Kubernetes, o projeto CoreDNS é hospedado pelo
{{< glossary_tooltip text="CNCF" term_id="cncf" >}}.

> Você pode usar o CoreDNS ao invés de kube-dns no seu cluster substituindo por
> kube-dns em um deployment existente, ou usando ferramentas como kubeadm
> que fará o deploy e o upgrade do cluster pra você.
> 
> ## Instalando o CoreDNS
> 
> Para deployment manual, ou substituição do kube-dns, veja a documentação no
> [CoreDNS website](https://coredns.io/manual/installation/).

Reviewer: Quando nos referimos ao objeto Deployment do Kubernetes é melhor manter a versão em inglês, mas neste caso como são verbos e se referem ao cluster em si, a leitura fica mais fluida se usarmos os termos existentes do português:

```suggestion
Você pode usar o CoreDNS ao invés de kube-dns no seu cluster substituindo por
kube-dns em uma implantação existente, ou usando ferramentas como kubeadm
que fará a instalação e a atualização do cluster pra você.

## Instalando o CoreDNS

Para instalação manual, ou substituição do kube-dns, veja a documentação no 
[site do CoreDNS](https://coredns.io/manual/installation/).
```

## Migrando para CoreDNS

### Atualizando um cluster existente com kubeadm

No Kubernetes versão 1.21, kubeadm removeu o suporte para  `kube-dns` como uma aplicação DNS.
Para `kubeadm` v{{< skew currentVersion >}}, o único Cluster DNS suportado é o CoreDNS.

Você pode migrar para o CoreDNS quando usar o  `kubeadm` para atualizar o cluster que está usando 
`kube-dns`. Neste caso, `kubeadm` gera a configuração do CoreDNS

> ("Corefile") basedo no `kube-dns` ConfigMap, preservando a configuração para 
> stub domains, e upstream name server.

Reviewer: Sugestão para melhor leitura (ConfigMap <nome> ao invés de <nome> ConfigMap) e uma correção pequena de erro tipográfico:
```suggestion
("Corefile") baseado no ConfigMap `kube-dns`, preservando a configuração para 
stub domains e upstream name server.
```

## Atualizando CoreDNS

> Você pode verificar a versão do CoreDNS que o kubeadm instala para cada versão do Kubernetes na pagina 

Reviewer: Faltou um acento em "página":

```suggestion
Você pode verificar a versão do CoreDNS que o kubeadm instala para cada versão do Kubernetes na página 
```

[versão do CoreDNS no Kubernetes](https://github.com/coredns/deployment/blob/master/kubernetes/CoreDNS-k8s_version.md).

CoreDNS pode ser atualizado manualmente, caso você queria atualizar somente o CoreDNS
ou usar sua própria imagem customizada.

> Aqui está um link útil [diretriz e passo a passo](https://github.com/coredns/deployment/blob/master/kubernetes/Upgrading_CoreDNS.md)
> disponivel para garantir uma atualização tranquilo.
> Certifique-se que a configuração existente do CoreDNS ("Corefile") é mantida quando atualizar o seu cluster.
> 
> Se você está atualizando o seu cluster usando a ferramenta `kubeadm`, O `kubeadm`
> pode cuidar da retenção da configuração existente do CoreDNS automaticamente.

Reviewer: Usualmente evitamos o uso de "aqui está um link..." ou "clique aqui", de acordo com o guia de estilo da documentação: https://kubernetes.io/pt-br/docs/contribute/style/style-guide/#links.

O outro ponto é que diretriz, no português, tem mais um sentido de normas do que de instruções. Usar instruções aqui seria mais adequado. Sugestão:

```suggestion
Há uma página de [instruções e passo-a-passo](https://github.com/coredns/deployment/blob/master/kubernetes/Upgrading_CoreDNS.md)
disponível para garantir uma atualização sem problemas.
Certifique-se que a configuração existente do CoreDNS ("Corefile") é mantida quando atualizar o seu cluster.

Se você está atualizando o seu cluster usando a ferramenta `kubeadm`, o `kubeadm`
pode cuidar da retenção da configuração existente do CoreDNS automaticamente.
```

## Ajustando o CoreDNS

Quando a utilização dos recursos é uma preocupação, pode ser útil ajustar a configuração do CoreDNS. Para mais detalhes, confira [documentação para escalonar o CoreDNS](https://github.com/coredns/deployment/blob/master/kubernetes/Scaling_CoreDNS.md).

## {{% heading "whatsnext" %}}

> Você pode configurar o [CoreDNS](https://coredns.io) para suportar muitos outros cenários de uso que o
> kube-dns, apenas por modificar a configuração do CoreDNS("Corefile").
> Para mais informações, veja a [documentação](https://coredns.io/plugins/kubernetes/)
> para o `kubernetes` CoreDNS plugin, ou leia o 
> [Entradas DNS personalizadas para Kubernetes](https://coredns.io/2017/05/08/custom-dns-entries-for-kubernetes/).
> no blog do CoreDNS.

Reviewer: Alguns pontos aqui:
* Acredito que "caso de uso" seja mais amplamente utilizado em engenharia de software no português do que "cenário de uso".
* A ideia do primeiro parágrafo é enfatizar que o CoreDNS suporta mais casos de uso do que o kube-dns suportava, dependendo da configuração efetuada. Porém, a adição do "apenas" dá uma ideia de simplicidade que não estava presente na documentação original.
* É legal manter um espaço entre o texto e o início dos parênteses para legibilidade.
* Frases como "CoreDNS plugin" ficam mais legíveis em português quando escritas na ordem natural (plugin do CoreDNS).
* Notei que o artigo referido do blog do CoreDNS não está traduzido. Nesse caso, seria mais interessante não traduzirmos o título, já que se trata de um link externo, e colocar uma informação ao fim de que o texto encontra-se em inglês.

Adicionei as sugestões no bloco abaixo:

```suggestion
Você pode configurar o [CoreDNS](https://coredns.io) para suportar mais casos de uso do que o
kube-dns suporta modificando a configuração do CoreDNS ("Corefile").
Para mais informações, veja a [documentação](https://coredns.io/plugins/kubernetes/)
do plugin `kubernetes` do CoreDNS, ou leia o artigo
[Custom DNS Entries For Kubernetes](https://coredns.io/2017/05/08/custom-dns-entries-for-kubernetes/) (em inglês) no blog do CoreDNS.
```

---
reviewers:

> - johnbelamaric

Reviewer 2: Reviewer name can be removed

title: Usando o CoreDNS para Descoberta de Serviços
min-kubernetes-server-version: v1.9
content_type: task
weight: 380
---

<!-- overview -->

> Essa página descreve o processo de upgrade de CoreDNS e como instalar o CoreDNS ao invés de kube-dns.

Reviewer 2: "atualização do" instead of "upgrade de"

## {{% heading "prerequisites" %}}

{{< include "task-tutorial-prereqs.md" >}} {{< version-check >}}

<!-- steps -->

## Sobre o CoreDNS

> [CoreDNS](https://coredns.io) é um servidor DNS flexivel e extesinvo

Reviewer 2: typo: flexivel and extesinvo

Maybe it will be better "é um servidor DNS flexível e extensível".

que pode servir como Kubernetes cluster DNS.
Como o Kubernetes, o projeto CoreDNS é hospedado pelo
{{< glossary_tooltip text="CNCF" term_id="cncf" >}}.

Você pode usar o CoreDNS ao invés de kube-dns no seu cluster substituindo por
kube-dns em um deployment existente, ou usando ferramentas como kubeadm
que fará o deploy e o upgrade do cluster pra você.

## Instalando o CoreDNS

Para deployment manual, ou substituição do kube-dns, veja a documentação no
[CoreDNS website](https://coredns.io/manual/installation/).

## Migrando para CoreDNS

### Atualizando um cluster existente com kubeadm

No Kubernetes versão 1.21, kubeadm removeu o suporte para  `kube-dns` como uma aplicação DNS.
Para `kubeadm` v{{< skew currentVersion >}}, o único Cluster DNS suportado é o CoreDNS.

Você pode migrar para o CoreDNS quando usar o  `kubeadm` para atualizar o cluster que está usando 
`kube-dns`. Neste caso, `kubeadm` gera a configuração do CoreDNS
("Corefile") basedo no `kube-dns` ConfigMap, preservando a configuração para 
stub domains, e upstream name server.

## Atualizando CoreDNS

Você pode verificar a versão do CoreDNS que o kubeadm instala para cada versão do Kubernetes na pagina 

> [CoreDNS versão no Kubernetes](https://github.com/coredns/deployment/blob/master/kubernetes/CoreDNS-k8s_version.md).

Reviewer 2: Sounds better to me "versão do CoreDNS no Kubernetes". What do you think?

CoreDNS pode ser atualizado manualmente, caso você queria atualizar somente o CoreDNS
ou usar sua própria imagem customizada.
Aqui está um link útil [diretriz e passo a passo](https://github.com/coredns/deployment/blob/master/kubernetes/Upgrading_CoreDNS.md)

> disponivel para garantir um upgrade tranquilo.

Reviewer 2: "uma atualização" instead of "um upgrade"

> Garanta que a configuração existente do CoreDNS ("Corefile") é mantida quando atualizar o seu cluster.

Reviewer 2: "Certifique-se que a configuração..."

> Se você está atualizando o seu cluster usando a ferramenta `kubeadm`, O`kubeadm`

Reviewer 2: typo: O `kubeadm`

pode cuidar da retenção da configuração existente do CoreDNS automaticamente.

## Ajustando o CoreDNS

> Quando a utilização dos recursos é uma preocupação, pode ser útil ajustar a configuração do CoreDNS. Para mais detalhes, confira [documentação para escalar o CoreDNS](https://github.com/coredns/deployment/blob/master/kubernetes/Scaling_CoreDNS.md).

Reviewer 2: Sounds better to me "documentação para escalonar" instead of "documentação para escalar".

## {{% heading "whatsnext" %}}

> Você pode configurar o [CoreDNS](https://coredns.io) para suportar muitos outros cenarios de uso que o

Reviewer 2: typo: cenarios
